util/topsql/reporter/reporter.go

```diff
@@ -302,15 +295,16 @@ func (tsr *RemoteTopSQLReporter) Collect(timestamp uint64, records []tracecpu.SQ
 
 // Close uses to close and release the reporter resource.
 func (tsr *RemoteTopSQLReporter) Close() {
-	tsr.dataSinks.mu.Lock()
-	defer tsr.dataSinks.mu.Unlock()
-
-	for d := range tsr.dataSinks.m {
-		d.CloseDataSink()
-		delete(tsr.dataSinks.m, d)
-	}
+	var m map[DataSink]struct{}
 
+	tsr.dataSinkMu.Lock()
+	m, tsr.dataSinks = tsr.dataSinks, make(map[DataSink]struct{})
 	tsr.cancel()
```

Comment:
I suppose you'd better move this line out. It has nothing related with the mutex.